app/src/app/job/hybrid.rs

```diff
@@ -331,17 +331,20 @@ impl JobApp for HybridJobAppImpl {
                         .enqueue_result_direct(id, data)
                         .await
                 }
-                Ok(ResponseType::ListenAfter) => {
+                Ok(rtype) => {
                     // publish for listening result client
-                    self.job_result_pubsub_repository()
-                        .publish_result(id, data)
-                        .await?;
+                    let r = self
+                        .job_result_pubsub_repository()
+                        .publish_result(id, data, rtype == ResponseType::ListenAfter)
```

Comment:
send result normally to result stream. Result for listen after is published only when responseType == ListenAfter (publish if last argument is true)